perf/reader.go

```diff
@@ -266,6 +268,20 @@ func (pr *Reader) Close() error {
 //
 // Calling Close interrupts the function.
 func (pr *Reader) Read() (Record, error) {
+	return pr.read(-1)
+}
+
+// ReadTimeout is Read but will timeout and return ErrNoRecords
+// if the timeout expires. The minimum timeout is 1 millisecond.
+func (pr *Reader) ReadTimeout(timeout time.Duration) (Record, error) {
+	if timeout < time.Millisecond {
```

Comment:
I'm not sure if we should enforce this. A timeout of 0 is valid and can be used for polling if the caller wants to do that.

What we should clearly document, though, is the following:
```
	ms := time.Millisecond
	fmt.Println(ms, ms.Milliseconds())  // 1ms 1
	ms -= time.Microsecond
	fmt.Println(ms, ms.Milliseconds()) // 999µs 0
	ms /= 2
	fmt.Println(ms, ms.Milliseconds()) // 499.5µs 0
```

Looks like it always gets rounded down, so perhaps we could error on `timeout > 0 && timeout < time.Millisecond` to prevent accidental busypolling?